### PR TITLE
LF-3377 Repeat crop plan spotlight

### PR DESCRIPTION
**Pull request by Tbrid**
**Description**

Added new spotlight modal when creating first repeating management plan. Along with new spotlight flags 

Jira link: https://lite-farm.atlassian.net/browse/LF-3377

**Type of change**

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [x] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Comment by iperdomo** [comment]
I tested the the highlighted area is off-place.
spotlight-click-here

File: packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const onFinish = () => dispatch(setSpotlightToShown('repeat_management_plan_creation'));
+
+  return (
+    <TourProviderWrapper
+      open={repeatingPlanCreated && !repeat_management_plan_creation}
+      steps={[
+        {
+          title: t('MANAGEMENT_PLAN.REPEATED_MP_SPOTLIGHT.TITLE'),
+          contents: [t('MANAGEMENT_PLAN.REPEATED_MP_SPOTLIGHT.BODY')],
+          selector: `#repeatPlan0`,
```
Comment (iperdomo) [inline packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx:21]: Off by one, there is no `#repeatPlan0`, the first one is `#repeatPlan1`

**Comment by iperdomo** [comment]
After using `#repeatCropPlan1` for the `selector` property, we get:
Screen Shot 2023-07-28 at 16 15 47

```diff
diff --git a/packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx b/packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx
index 3d59e5a83..6b4086792 100644
--- a/packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx
+++ b/packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx
@@ -18,7 +18,7 @@ export default function RepeatedCropPlanSpotlight({ repeatingPlanCreated, childr
         {
           title: t('MANAGEMENT_PLAN.REPEATED_MP_SPOTLIGHT.TITLE'),
           contents: [t('MANAGEMENT_PLAN.REPEATED_MP_SPOTLIGHT.BODY')],
-          selector: `#repeatPlan0`,
+          selector: `#repeatPlan1`,
           position: 'center',
           offset: '1px',
           buttonText: t('common:GOT_IT'),

```

File: packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx (hunk 1 above)
Comment (Tbrid) [reply packages/webapp/src/containers/Crop/RepeatedCropPlanSpotlight/index.jsx]: Fixed with just labeling if it's a repeating plan.

**Comment by iperdomo** [comment]
After the most recent changes 1366x768
Screen Shot 2023-07-29 at 23 37 30

**Comment by iperdomo** [comment]
Portrait 768x1366
Screen Shot 2023-07-30 at 06 32 19

Mobile 360x800 - the indicator is _off_
Screen Shot 2023-07-30 at 06 35 00

**Comment by iperdomo** [comment]
Would it be possible to fix the mobile (portrait) indicator? is off 
Screen Shot 2023-07-30 at 06 35 00

**Comment by Tbrid** [comment]
> Would it be possible to fix the mobile (portrait) indicator? is off Screen Shot 2023-07-30 at 06 35 00

I used the position 'top' for the popover to make it so that it doesn't block the highlighted area, which would happen sometimes if the plan happens to be in the middle. Added an extra field to make this indicator optional on the popover.

**Comment by iperdomo** [comment]
@Tbrid thanks for the work. Can you please merge the most recent changes from `integration` in your branch? Thanks

File: packages/webapp/src/components/Crop/Management.jsx [hunk 2]
```diff
@@ -96,6 +96,11 @@ export default function PureCropManagement({
       {managementPlanCardContents && (
         <CardWithStatusContainer style={{ paddingTop: '16px' }}>
           {filteredManagementPlanCardContents.map((managementPlan, index) => {
+            if (index > 1 && index < 5) {
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Crop/Management.jsx:99]: Is this a testing artefact? 

Causing some weird behaviours by forcing index:

Screenshot 2023-07-31 at 10 20 41 AM

**Review by Duncan-Brain — CHANGES_REQUESTED** [review]
Migration works. And it seems like most things are working right.

If you could fix setting that index manually -- it is breaking some things for me. 

As Ivan mentioned -  I think merging integration will allow full feature set to be tested in conjunction.

File: packages/webapp/src/components/Crop/Management.jsx (hunk 2 above)
Comment (Tbrid) [reply packages/webapp/src/components/Crop/Management.jsx]: Nice catch. Removed.

**Review by Duncan-Brain — CHANGES_REQUESTED** [review]
Awesome, Found this too sorry I didn't catch it last time - left hand border is sticking to the screen on mobile views.

Screenshot 2023-07-31 at 12 06 32 PM

**Review by Duncan-Brain — APPROVED** [review]
@Tbrid Thanks so much looks great!

**Pull request merged**